server/mailet/integration-testing/src/test/java/org/apache/james/mailets/configuration/SmtpConfiguration.java

```diff
@@ -0,0 +1,102 @@
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailets.configuration;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.StringReader;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Optional;
+
+import org.apache.commons.io.IOUtils;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.google.common.base.Preconditions;
+
+public class SmtpConfiguration implements SerializableAsXml {
+    public static final boolean AUTH_REQUIRED = true;
+    public static final SmtpConfiguration DEFAULT = SmtpConfiguration.builder().build();
+
+    public static class Builder {
+        private Optional<Boolean> authRequired;
+        private Optional<String> authorizedAddresses;
+
+        public Builder() {
+            authorizedAddresses = Optional.empty();
+            authRequired = Optional.empty();
+        }
+
+        public Builder withAutorizedAddresses(String authorizedAddresses) {
+            Preconditions.checkNotNull(authorizedAddresses);
+            this.authorizedAddresses = Optional.of(authorizedAddresses);
+            return this;
+        }
+
+        public Builder requireAuthentication() {
+            this.authRequired = Optional.of(AUTH_REQUIRED);
+            return this;
+        }
+
+        public SmtpConfiguration build() {
+            return new SmtpConfiguration(authorizedAddresses, authRequired.orElse(!AUTH_REQUIRED));
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private final Optional<String> authorizedAddresses;
+    private final boolean authRequired;
+
+    public SmtpConfiguration(Optional<String> authorizedAddresses, boolean authRequired) {
+        this.authorizedAddresses = authorizedAddresses;
+        this.authRequired = authRequired;
+    }
+
+    public String serializeAsXml() throws IOException {
```

Comment:
that's crazy : instead of removing xml requirement, we add another layer to generate xml configuration.
Am I the only one wanting things to be as simple as possible ?

Reply:
That is exactly what has been done for **mailetcontainer.xml**.

Do you have a better configuration generation strategy?

Like:
 - Introduce a SmtpConfiguration POJO object that SmtpServerFactory accept directly accept
 - Generate it from the actual XML file 
 - Inject it

Most probably fighting with Spring is part of that work.

I would call that a heavy refactoring. It implies redefining how we do configuration. What is a major design change. And I don't like doing this in a **add integration test PR**. I think it's normal that I try to do without, then discuss with my collegues what they actually think of the current solution, finally, later on, move to a more *change breaker* implementation if needed. I don't think I'm **crazy**.

Reply:
:+1: with @mbaechler but also :+1: with @chibenwa it is probably to much work in this PR but in this case WE MUST CREATE A TICKET about that

Reply:
As written to @mbaechler : 

```
(translated from french)
Benoît STG_TELLIER
    Now we use a builder for SMTP XML conf
    Tomorrow we will use another builder for a better higher level SMTP conf object
    Both will have close parameters, as they represent the same objects
Did I introduce technical debt we can not pay? Or does a "builder switch" seems doable when time come?

Matthieu EXT_BAECHLER
You're right, I'm OK with approach.
```